internal/static/integrations/windows_exporter/windows_exporter_windows.go

```diff
@@ -29,11 +36,22 @@ func New(logger log.Logger, c *Config) (integrations.Integration, error) {
 		return nil, err
 	}
 
-	return integrations.NewCollectorIntegration(c.Name(), integrations.WithCollectors(
-		// Hard-coded 4m timeout to represent the time a series goes stale.
-		// TODO: Make configurable if useful.
-		collector.NewPrometheus(4*time.Minute, &winCol, logger),
-	)), nil
+	return integrations.NewCollectorIntegration(
+		c.Name(),
+		integrations.WithCollectors(
+			// Hard-coded 4m timeout to represent the time a series goes stale.
+			// TODO: Make configurable if useful.
+			collector.NewPrometheus(4*time.Minute, &winCol, logger),
+		),
+		integrations.WithRunner(func(ctx context.Context) error {
+			<-ctx.Done()
+
+			// Stop the collector
+			err := winCol.Close()
+
+			return errors.Join(ctx.Err(), err)
+		}),
```

Comment:
Could you please take a look here, if this is the right place to add an Close method?

Reply:
Yes, though there is possible a slight difference from a normal component. during a an Update operation this will be closed and reopened. So has long as this doesnt prevent reloads then we are fine.

https://github.com/grafana/alloy/blob/2ab9e0d1d40b7e25612b50fe06e2dfd1d580a339/internal/component/prometheus/exporter/exporter.go#L78

Reply:
I’m planning to move away from using the global scope. It has been indicated that some handles remain open if they are used in this way. Instead, the handles should be opened in the initialization functions.

For the exporter as a standalone binary, this isn’t as critical. However, for binaries that embed the exporter, it's important to close handles to avoid handle leaks.